File: src/linter/diagnostic.rs
use serde::Serialize;

#[derive(Serialize, Debug)]
pub struct Position {
    pub line: u32,
    pub character: u32,
}

#[derive(Serialize, Debug)]
pub struct Range {
    pub source: String,
    pub start: Position,
    pub end: Position,
}

#[derive(Serialize, Debug)]
pub struct Diagnostic {
    pub severity: String,
    pub code: u16,
    pub ranges: Vec<Range>,
    pub description: String,
}

File: action/src/typings/diagnostic.ts
import { z } from "zod";

const positionSchema = z.object({
  line: z.number().min(0),
  character: z.number().min(0),
});

export const fragmentSchema = z.object({
  source: z.string(),
  start: positionSchema,
  end: positionSchema,
});

export const diagnosticSchema = z.object({
    code: z.number().min(0),
    severity: z.string(),
    ranges: z.array(fragmentSchema),
    description: z.string(),
})

export type Fragment = z.infer<typeof fragmentSchema>
export type Diagnostic = z.infer<typeof diagnosticSchema>